common/SigUtil.cpp

```diff
@@ -150,6 +150,15 @@ void requestShutdown()
     /// Close the signalLog file.
     void signalLogClose()
     {
+        // We cannot shutdown the logging subsystem
+        // because freeing memory is not signal safe.
+
+        // Flush the IO buffers.
+        std::flush(std::cout);
```

Comment:
Hmm - surely std::cout is yet another layer of C++ buffered-ness on top of what we have, and will no doubt allocate memory, and is not on the list of signal safe methods for good reason. Why are we using them? we should write to stdout or stderr - using safe, low-level system calls - not the FILE* abstraction, and not the C++ wrappers :-) cf. https://stackoverflow.com/questions/21460620/is-fflush-safe-to-be-called-from-a-signal-handler

Reply:
Logging does _not_ use std::cout/cerr. This is just flushing. Technically, since we do flush stdout/stderr, and since cout and cerr are really mapped to them, this shouldn't be needed. I just put them for completeness. I'll remove them.

Reply:
Great; please do cut those out. We shouldn't write to stdout/stderr either but the relevant file-descriptions.

Reply:
Removed.